lib/Type.lua

```diff
@@ -0,0 +1,30 @@
+--[[
+	Contains markers for annotating objects with types.
+
+	To set the type of an object, use `Type` as a key and the actual marker as
+	the value:
+
+		local foo = {
+			[Type] = Type.Foo,
+		}
+]]
+
+local Symbol = require(script.Parent.Symbol)
+
+local Type = newproxy(true)
```

Comment:
That's a good idea, it would make the typechecks more in line with those using `typeof` as well.

I don't think I want it to return any types except internal Roact ones, since for those you can just use `typeof` and be more clear.